src/mpls.rs

```diff
@@ -0,0 +1,277 @@
+        // lbl_tc_s should become 0b1110 (E from label) | 0b1111 (preserved TC+S) = 0xEF
+        assert_eq!(mpls_header2.label(), 0xABCDE);
+        assert_eq!(mpls_header2.tc(), 0x07); // Preserved
+        assert_eq!(mpls_header2.get_s(), 0x01); // Preserved
+        assert_eq!(mpls_bytes, [0xAB, 0xCD, 0xEF, 0xFF]);
+    }
+
+    #[test]
+    fn test_mpls_set_tc() {
+        let mut mpls_bytes: [u8; Mpls::LEN] = [0x00, 0x00, 0xA5, 0x00]; // lbl_tc_s = 0b10100101 (L=A, TC=2, S=1)
+        let mpls_header = unsafe { mpls_from_bytes_mut(&mut mpls_bytes) };
+
+        mpls_header.set_tc(0x06); // Set TC to 6 (0b110)
+        // Expected lbl_tc_s: L=A (0b1010....), new TC=6 (0b..0110..), S=1 (0b.......1)
+        // lbl_tc_s should be 0b10101101 = 0xAD
+        assert_eq!(mpls_header.tc(), 0x06);
+        assert_eq!(((mpls_bytes[2] & Mpls::LABEL_LOWER_MASK_IN_BYTE) >> Mpls::LABEL_LOWER_SHIFT), 0x0A); // Label part preserved
+        assert_eq!((mpls_bytes[2] & Mpls::S_MASK_IN_BYTE), 0x01); // S bit preserved
+        assert_eq!(mpls_bytes, [0x00, 0x00, 0xAD, 0x00]);
+    }
+
+    #[test]
+    fn test_mpls_set_s() {
+        let mut mpls_bytes: [u8; Mpls::LEN] = [0x00, 0x00, 0xA6, 0x00]; // lbl_tc_s = 0b10100110 (L=A, TC=3, S=0)
+        let mpls_header = unsafe { mpls_from_bytes_mut(&mut mpls_bytes) };
+
+        mpls_header.set_s(0x01); // Set S to 1
+        // Expected lbl_tc_s: L=A (0b1010....), TC=3 (0b..0011..), new S=1 (0bاريات1)
+        // lbl_tc_s should be 0b10100111 = 0xA7
+        assert_eq!(mpls_header.get_s(), 0x01);
+        assert_eq!(((mpls_bytes[2] & Mpls::LABEL_LOWER_MASK_IN_BYTE) >> Mpls::LABEL_LOWER_SHIFT), 0x0A); // Label part preserved
+        assert_eq!(((mpls_bytes[2] & Mpls::TC_MASK_IN_BYTE) >> Mpls::TC_SHIFT), 0x03); // TC preserved
+        assert_eq!(mpls_bytes, [0x00, 0x00, 0xA7, 0x00]);
+    }
+
+    #[test]
+    fn test_mpls_set_ttl() {
+        let mut mpls_bytes: [u8; Mpls::LEN] = [0x12, 0x34, 0x56, 0x00];
+        let mpls_header = unsafe { mpls_from_bytes_mut(&mut mpls_bytes) };
+
+        mpls_header.set_ttl(0xFF);
+        assert_eq!(mpls_header.ttl(), 0xFF);
+        assert_eq!(mpls_bytes, [0x12, 0x34, 0x56, 0xFF]);
+    }
+
+    #[test]
+    fn test_mpls_combined_setters_and_getters() {
+        let mut mpls_bytes: [u8; Mpls::LEN] = [0x00; Mpls::LEN];
+        let mpls_header = unsafe { mpls_from_bytes_mut(&mut mpls_bytes) };
+
+        mpls_header.set_label(0x98765);
+        mpls_header.set_tc(0x03);
+        mpls_header.set_s(0x01);
+        mpls_header.set_ttl(0xAA);
+
+        // First verify all getter values
+        assert_eq!(mpls_header.label(), 0x98765);
+        assert_eq!(mpls_header.tc(), 0x03);
+        assert_eq!(mpls_header.get_s(), 0x01);
+        assert_eq!(mpls_header.ttl(), 0xAA);
+
+        // Drop the mutable borrow before comparing bytes according to compiler warning
+        let _ = mpls_header;
```

Comment:
I think using

```suggestion
        drop(mpls_header);
```

would be more more understandable.

Reply:
This is already fixed.